File: android/app/src/main/java/de/davidgrieser/container/model/KioskConfig.kt
package de.davidgrieser.container.model

/**
 * A single web app that the container is allowed to display.
 *
 * @param name  Human-readable label shown in the menu.
 * @param iconUrl Absolute https URL of the app icon (optional).
 * @param url   Absolute https URL of the page to display. Navigation is locked
 *              to this URL's registered domain.
 */
data class AppEntry(
    val name: String,
    val iconUrl: String?,
    val url: String
)

/** The full remotely-controlled configuration. */
data class KioskConfig(
    val apps: List<AppEntry>
) {
    val isEmpty: Boolean get() = apps.isEmpty()
}

File: ios/ContainerKit/Sources/ContainerKit/Core/KioskConfig.swift
import Foundation

/// A single web app the container is allowed to display.
public struct AppEntry: Equatable, Sendable {

    /// Human-readable label shown in the menu.
    public let name: String

    /// Absolute https URL of the app icon, if the configuration gave one.
    public let iconURL: String?

    /// Absolute https URL of the page. Navigation is locked to this URL's
    /// registered domain while it is showing.
    public let url: String

    public init(name: String, iconURL: String?, url: String) {
        self.name = name
        self.iconURL = iconURL
        self.url = url
    }
}

/// The full remotely-controlled configuration.
public struct KioskConfig: Equatable, Sendable {

    public let apps: [AppEntry]

    public init(apps: [AppEntry]) {
        self.apps = apps
    }

    public var isEmpty: Bool { apps.isEmpty }

    public static let empty = KioskConfig(apps: [])
}
